sklearn/linear_model/tests/test_omp.py

```diff
@@ -30,6 +32,34 @@
 # and y (n_samples, 3)
 
 
+# FIXME: 'normalize' to be removed in 1.2
+@pytest.mark.parametrize('OmpModel', [
+    OrthogonalMatchingPursuit,
+    OrthogonalMatchingPursuitCV
+])
+@pytest.mark.parametrize(
+    'normalize, n_warnings',
+    [(True, 0),
+     (False, 0),
+     ("deprecated", 1)]
+)
+def test_assure_warning_when_normalize(OmpModel, normalize, n_warnings):
+    # check that we issue a FutureWarning when normalize was set
+    rng = check_random_state(0)
+    n_samples = 200
+    n_features = 2
+    X = rng.randn(n_samples, n_features)
+    X[X < 0.1] = 0.
+    y = rng.rand(n_samples)
+
+    model = OmpModel(normalize=normalize)
+    with warnings.catch_warnings(record=True) as record:
```

Comment:
Base on [this example in the doc of pytest](https://docs.pytest.org/en/6.2.x/warnings.html#recwarn).
```suggestion
    with pytest.warns(None) as record:
```